File: src/java/com/wanhive/basic/web/Navigator.java
/**********************************************************
 * NAVIGATOR: Navigation Bar, Help Button
 * Copyright (C) 2009, 2010  Amit Kumar(amitkriit@gmail.com)
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 ***********************************************************/
package com.wanhive.basic.web;

import java.io.IOException;
import java.io.PrintWriter;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

import com.wanhive.basic.beans.User;
import com.wanhive.basic.db.MenuManager;


/**
 * Servlet implementation class Navigator
 */
public class Navigator extends HttpServlet {
	private static final long serialVersionUID = 1L;
       
    /**
     * @see HttpServlet#HttpServlet()
     */
    public Navigator() {
        super();
    }

	/**
	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
	 */
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		PrintWriter out=response.getWriter();
		response.setContentType("text/html");
		String path=request.getParameter("path");
		int id;
		try{id=Integer.parseInt(path);}catch (Exception e) {id=1;}
		HttpSession session=request.getSession();
		User user=(User)session.getAttribute("currentUser");
		if(user==null)
		{
			out.println("Session Error: <a href='index.jsp'>Please Login</a>");
			session.invalidate();
		}
		else
		{
			String responseStr=MenuManager.breadcrumb(id,user.getId());
			out.println(responseStr);
		}
	}

	/**
	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
	 */
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		doGet(request, response);
	}

}

File: web/themes/js/uncompressed/menu/navigator.js
/**********************************************************
 * Renders the navigation bar with breadcrumb, refresh button and help button(Main page)
 * Copyright (C) 2009  Amit Kumar(amitkriit@gmail.com)
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 ***********************************************************/


var myNavigatorRequest=getHTMLHTTPRequest();

function callNavigatorAjax(path)
{
	myMoveAction.reInit();				//Defined in globalvars.js
	internalWindow.closeAll();
	//Ref: helpWindowOpener.js
	myHelpWindowReloadState=true;	//After reload help window must be reloaded
	var myRandom=parseInt(Math.random()*99999999);
	var url=myServerAddress+"Navigator?rand="+myRandom+"&path="+path;
	myNavigatorRequest.open("GET",url,true);

	/* This function processes the response */
	myNavigatorRequest.onreadystatechange=drawNavAction;

	/* Send a new request */
	writeWaitMsg(div,"themes/icons/ajax_loading/8.gif","");
	myNavigatorRequest.send(null);
	//repeatedAjaxRequest=true;
}

var div="breadcrumb";	//Ref: main.jsp

function drawNavAction() {
	if(myNavigatorRequest.readyState==4) {
		if(myNavigatorRequest.status==200) {
			var navText='&nbsp;Navigation: <font color="#345678"><i>'+myNavigatorRequest.responseText+'</i></font>';
			var managedText='<table cellpadding="0" cellspacing="0" border="1" width=100%><tr><td>'+navText+'</td></tr></table>';
			drawNav(myNavigatorRequest.responseText,div);
		}
		else {
			alert("Connection Problem:"+myNavigatorRequest.statusText);
		}
	}
}

function drawNav(navText,area) {
	document.getElementById(area).innerHTML=null;
	document.getElementById(area).innerHTML=navText;
}

/* Execute Script embedded inside div's innerHTML */
/*function executeCustomReload() {
	var x = document.getElementById(div).getElementsByTagName("script");    for(var i=0;i<x.length;i++)   {       eval(x[i].text);   }
}*/